### PR TITLE
Adding comment on the source tables to capture that they have been deprecated

### LINKED ISSUE #2375: [FEATURE]: Add comments to indicate that a Hive table has been migrated
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

Leverage [Databricks' capability to suggest a fix for Hive tables that are migrated to UC](https://docs.databricks.com/en/data-governance/unity-catalog/migrate.html#add-comments-to-indicate-that-a-hive-table-has-been-migrated) by adding a comment to the Hive table that is migrated:

> This table is deprecated. Please use catalog.default.table instead of hive_metastore.schema.table.

### PR DESCRIPTION
**Pull request by aminmovahed-db**
### Linked issues

Resolves #2375

### Functionality

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [x] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [x] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

**Comment by aminmovahed-db** [comment]
These are the screenshots of this feature...
image

image

**Review by asnare — COMMENTED** [review]
From a design perspective: Should this modify any existing comment on the table (by prepending?), rather than just overwriting it?

I'd really like to see an integration test.

**Review by nfx — APPROVED** [review]
lgtm

File: src/databricks/labs/ucx/hive_metastore/table_migrate.py [hunk 1]
```diff
@@ -402,6 +406,9 @@ def _init_seen_tables(self):
     def _sql_alter_to(self, table: Table, target_table_key: str):
         return f"ALTER {table.kind} {escape_sql_identifier(table.key)} SET TBLPROPERTIES ('upgraded_to' = '{target_table_key}');"
 
+    def _sql_add_comment(self, table: Table, target_table_key: str):
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/hive_metastore/table_migrate.py:409]: Make it more explicit which comment is added

```suggestion
    def _sql_add_deprecated_comment(self, table: Table, target_table_key: str):
        """Docs: https://docs.databricks.com/en/data-governance/unity-catalog/migrate.html#add-comments-to-indicate-that-a-hive-table-has-been-migrated"""
    
```

**Review by JCZuurmond — CHANGES_REQUESTED** [review]
One minor detail

File: src/databricks/labs/ucx/hive_metastore/table_migrate.py [hunk 2]
```diff
@@ -402,6 +406,10 @@ def _init_seen_tables(self):
     def _sql_alter_to(self, table: Table, target_table_key: str):
         return f"ALTER {table.kind} {escape_sql_identifier(table.key)} SET TBLPROPERTIES ('upgraded_to' = '{target_table_key}');"
 
+    def _sql_add_deprecated_comment(self, table: Table, target_table_key: str):
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/hive_metastore/table_migrate.py:409]: Sorry, rereading the link in the docs. The following is probably better:

```suggestion
    def _sql_add_migrated_comment(self, table: Table, target_table_key: str):
```

**Review by JCZuurmond — APPROVED** [review]
one comment, otherwise lgtm

**Review by JCZuurmond — CHANGES_REQUESTED** [review]
Comment on view does not work

**Pull request merged**